src/full/Agda/Interaction/Highlighting/Range.hs

```diff
@@ -62,6 +63,9 @@ overlapping :: Range -> Range -> Bool
 overlapping r1 r2 = not $
   to r1 <= from r2 || to r2 <= from r1
 
+overlappings :: Ranges -> Ranges -> Bool
+overlappings (Ranges r1s) (Ranges r2s) = or [ overlapping r1 r2 | r1 <- r1s, r2 <- r2s ]
```

Comment:
Since the ranges inside `Ranges` are ordered ascendingly, there should be a linear implementation rather than a quadratic one.